### PR TITLE
DM-18745: VisitAnalysis "distance" from refCat plots don't appear to use jointcal

### PR DESCRIPTION
**Pull request by laurenam**

### CODE REVIEW

File: python/lsst/pipe/analysis/utils.py [hunk 1]
```diff
@@ -222,6 +223,30 @@ def __call__(self, catalog):
         cosDec2 = np.cos(catalog[self.declination2]) if self.declination2 is not None else 1.0
         return (first*cosDec1 - second*cosDec2)*(1.0*afwGeom.radians).asArcseconds()*self.unitScale
 
+class AngularDistance(object):
+    """Functor to calculate the Haversine angular distance between two points
```
Comment (yalsayyad) [inline python/lsst/pipe/analysis/utils.py:227]: This docstring confused me.  This really computes the distance between two columns right?

File: python/lsst/pipe/analysis/coaddAnalysis.py [hunk 2]
```diff
@@ -964,7 +962,7 @@ def plotMatches(self, matches, filterName, filenamer, dataId, description="match
                                    cat["distance"]*(1.0*afwGeom.radians).asArcseconds()*self.unitScale,
                                "Distance (%s) (calib_astrom_used)" % unitStr, shortName,
                                self.config.analysisMatches, prefix="src_", goodKeys=["calib_astrometry_used"],
-                               qMin=-0.01*matchRadius, qMax=0.5*matchRadius,
+                               qMin=-0.01*self.matchRadius, qMax=0.5*self.matchRadius,
```
Comment (yalsayyad) [inline python/lsst/pipe/analysis/coaddAnalysis.py]: I don't follow why qMin and qMax are based on `self.matchRadius` instead of `matchRadius`

**Review by yalsayyad — APPROVED** [review]

File: python/lsst/pipe/analysis/utils.py (hunk 1 above)
Comment (laurenam) [reply python/lsst/pipe/analysis/utils.py:227]: Well, from Wikipedia:  "The haversine formula determines the great-circle distance between two points on a sphere given their longitudes and latitudes."  I do initialize the class using 4 columns (with names `raStr1`, `decStr1`, `raStr2`, `decStr2`) that represent the two points between which we want to compute this distance.  I've updated the docstring to make this more clear.

File: python/lsst/pipe/analysis/coaddAnalysis.py (hunk 2 above)
Comment (laurenam) [reply python/lsst/pipe/analysis/coaddAnalysis.py]: Ah, that was part of my removal of the erroneous matchRadius being plotted to the reference-matched plots (the matching was actually done in the stack in this case.  Here, we are just reading in those matches.)  I accomplish this by allowing for a `matchRadius` to be `None`, but still wanted a good guess at a good scaling for the y-axis and the "best guess" I have is the default for when we do our own matching (this does assume the two cases will have had roughly the same tolerance...but some dynamic tweaking will happen if the limits are not satisfactory according to the list of "criteria" in the `Analysis` class).

I do agree that this should be set to `matchRadius` if one is passed in, so I now do the scaling by a new variable defined as:
`qMatchScale = matchRadius if matchRadius else self.matchRadius`

**Pull request merged**